### PR TITLE
Elide secrets in alertmanager config

### PR DESCRIPTION
**Pull request by Conorbro**
Implements #840

### CODE REVIEW

File: config/testdata/conf.good.yml [hunk 1]
```diff
@@ -0,0 +1,125 @@
+global:
```
Comment (brian-brazil) [inline config/testdata/conf.good.yml]: Also slack's api url, pushover's token, opsgenic api_key, victorops api_key,

Comment (Conorbro) [inline config/testdata/conf.good.yml]: They've been added now to the bottom of the yml.

File: config/config_test.go [hunk 2]
```diff
@@ -60,3 +62,20 @@ receivers:
...
+	c, err := LoadFile("testdata/conf.good.yml")
+	if err != nil {
+		t.Errorf("Error parsing %s: %s", "testdata/good.yml", err)
+	}
+
+	// String method must not reveal authentication credentials.
+	s := c.String()
+	secretRe := regexp.MustCompile("<secret>")
+	matches := secretRe.FindAllStringIndex(s, -1)
+	if len(matches) != 18 || strings.Contains(s, "mysecret") {
```
Comment (brian-brazil) [inline config/config_test.go]: I count 10, how are you getting 18?

Comment (Conorbro) [inline config/config_test.go]: Remarshalling adds in fields that have secrets omitted.

Comment (brian-brazil) [inline config/config_test.go]: We should `omit_empty` any of those that are optional then.

File: config/config.go [hunk 3]
```diff
@@ -28,14 +28,18 @@ import (
...
 
-var patAuthLine = regexp.MustCompile(`((?:api_key|service_key|api_url|token|user_key|password|secret):\s+)(".+"|'.+'|[^\s]+)`)
-
 // Secret is a string that must not be revealed on marshaling.
 type Secret string
 
 // MarshalYAML implements the yaml.Marshaler interface.
 func (s Secret) MarshalYAML() (interface{}, error) {
-	return "<hidden>", nil
+	return "<secret>", nil
```
Comment (brian-brazil) [inline config/config.go]: Return empty if the actual value is empty

**Pull request merged**